src/courseware/course/course-license/CourseLicense.jsx

```diff
@@ -0,0 +1,161 @@
+
+  // Set the defaultVersion to 4.0
+  const defaultVersion = '4.0';
+  optionStr.split(' ').forEach(option => {
+    // Split the option into key and value
+    // Default the value to `true` if no value
+    let key = '';
+    let value = '';
+    if (option.indexOf('=') !== -1) {
+      [key, value] = option.split('=');
+    } else {
+      key = option;
+      value = true;
+    }
+
+    // Check for version
+    if (key === 'ver') {
+      version = value;
+    } else {
+      // Set the option key to lowercase to make
+      // it easier to query
+      options[key.toLowerCase()] = value;
+    }
+  });
+
+  // No options
+  if (Object.keys(options).length === 0) {
+    // If no other options are set for the
+    // license, set version to 1.0
+    version = '1.0';
+
+    // Set the `zero` option so the link
+    // works correctly
+    options = {
+      zero: true,
+    };
+  }
+
+  // Set the version to whatever was included,
+  // using `defaultVersion` as a fallback if unset
+  version = version || defaultVersion;
+
+  return [licenseType, options, version];
+}
+
+function CourseLicense({
+  license,
+  intl,
+}) {
+  const renderAllRightsReservedLicense = () => (
+    <div>
+      <FontAwesomeIcon aria-hidden="true" className="mr-1" icon={faCopyright} />
+      <span className="license-text">
+        {intl.formatMessage(messages['learn.course.license.allRightsReserved.text'])}
+      </span>
+    </div>
+  );
+
+  const renderCreativeCommonsLicense = (activeCreativeCommonsLicenseTags, version) => (
+    <a
+      rel="license noopener noreferrer"
+      target="_blank"
+      href={`https://creativecommons.org/licenses/${activeCreativeCommonsLicenseTags.join('-')}/${version}/`}
+    >
+      <span className="sr-only">
+        {intl.formatMessage(messages['learn.course.license.creativeCommons.terms.preamble'])}&nbsp;
+      </span>
+      <FontAwesomeIcon aria-hidden="true" className="mr-1" icon={faCreativeCommons} />
+      {activeCreativeCommonsLicenseTags.map(tag => (
+        <span key={tag}>
+          <span className="sr-only">
+            {intl.formatMessage(messages[CreativeCommonsLicenseTags[tag].intlMessagesId])}&nbsp;
+          </span>
+          <FontAwesomeIcon aria-hidden="true" className="mr-1" icon={CreativeCommonsLicenseTags[tag].icon} />
+        </span>
+      ))}
+      <span className="license-text">
+        {intl.formatMessage(messages['learn.course.license.creativeCommons.text'])}
+      </span>
+    </a>
+  );
+
+  const [licenseType, licenseOptions, licenseVersion] = parseLicense(license);
+
+  return (
+    <div className="course-license">
+      {licenseType === 'all-rights-reserved' && renderAllRightsReservedLicense()}
+      {licenseType === 'creative-commons' && renderCreativeCommonsLicense(
+        Object.keys(licenseOptions),
+        licenseVersion,
+      )}
+    </div>
+  );
+}
+
+CourseLicense.propTypes = {
+  license: PropTypes.string.isRequired,
+  intl: intlShape.isRequired,
+};
```

Comment:
This is going to work, but will show an error in the console. Wouldn't it be better to use a default value like this instead of checking `!license` in `parseLicense`?
```suggestion
CourseLicense.propTypes = {
  license: PropTypes.string,
  intl: intlShape.isRequired,
};

CourseLicense.defaultProps = {
  license: 'all-rights-reserved',
};
```

However this means that we need to create this component this way, because the API would return `null`, not `undefined`.
```js
<CourseLicense license={course.license || undefined} />
```